### PR TITLE
ceph-volume: adds support for the 'lvm batch' command

### PR DESCRIPTION
**Pull request by andrewschoen**
This feature is currently in development for ``ceph-volume`` and we need support in ``ceph-ansible`` to start building our functional tests.

I've added a test scenario here as well, but don't want to enable it for pull requests until this feature is fully complete in ``ceph-volume``. I've verified the scenario does pass though: https://2.jenkins.ceph.com/job/ceph-ansible-scenario/302/consoleFull

### CODE REVIEW

File: docs/source/osds/scenarios.rst [hunk 1]
```diff
@@ -187,10 +187,19 @@ is only available when the ceph release is Luminous or newer.
    The creation of the logical volumes is not supported by ``ceph-ansible``, ``ceph-volume``
    only creates OSDs from existing logical volumes.
 
-``lvm_volumes`` is the config option that needs to be defined to configure the
-mappings for devices to be deployed. It is a list of dictionaries which expects
-a volume name and a volume group for logical volumes, but can also accept
-a partition in the case of ``filestore`` for the ``journal``.
+``lvm_volumes`` or ``devices`` are the config option that needs to be defined to deploy OSDs
```
Comment (alfredodeza) [inline docs/source/osds/scenarios.rst]: I think this needs a sub-section, one for each of the config options:

    Configurations
    ^^^^^^^^^^^^^^^
    ....

    - ``lvm_volumes`` is a list of [...]

    - ``devices`` is a list of raw device names [...]

That way the separation is clear, and there is no need to have to read this sentence to grasp that there is more than one way to configure this

File: docs/source/osds/scenarios.rst [hunk 2]
```diff
@@ -187,10 +187,19 @@ is only available when the ceph release is Luminous or newer.
...
+``lvm_volumes`` is a list of dictionaries which expects a volume name and a volume group for
+logical volumes, but can also accept a partition in the case of ``filestore`` for the ``journal``.
+If ``lvm_volumes`` is defined then the ``ceph-volume lvm create`` command is used to create each OSD
+defined in ``lvm_volumes``.
+
+``devices`` is a list of raw device names as strings. If ``devices`` is defined then the ``ceph-volume lvm batch``
+command will be used to deploy OSDs.
+
+Both ``lvm_volumes`` and ``devices`` can be defined and both methods would be used in the deployment or you
+can pick just one method.
```
Comment (alfredodeza) [inline docs/source/osds/scenarios.rst]: I am on the fence with this one, is it useful to allow both? From the ceph-ansible side of things, is it easier to just allow them vs. picking one or the other?

Comment (andrewschoen) [reply docs/source/osds/scenarios.rst]: Why not allow both? It is easier from the ceph-ansible side to allow them both. It means less conditions in the ``when`` clause anyway when including the scenario files.

Comment (alfredodeza) [reply docs/source/osds/scenarios.rst]: Yeah we can allow that, was just surprised to see both.

**Comment by alfredodeza** [comment]
This branch allowed me to test the new filestore functionality to support `ceph-volume lvm batch` correctly. It would be nice if we could merge this so that we can start testing this in our CI

**Comment by leseb** [comment]
@alfredodeza what was so urgent in merging this? The PR was only opened two days ago. @guits and I are busy working on urgent PRs for finalising stable-3.1.
Next time, please at least ask us or tell us your priority/severity of your PRs before merging, so we can prioritize the merges.
Because right now, all our branches are not up to date and need rebase, which will re-trigger the CI, so basically we lost 4 hours with this. We were hoping merging PRs in that timeframe. Particularly this one  https://github.com/ceph/ceph-ansible/pull/2991 which has the highest severity at the moment.

**Comment by alfredodeza** [comment]
@leseb I realize you are busy, I was going by our agreement that we are being responsible for the ceph-volume component, this is blocking all the batch support that is already in master and we don't want to backport until this lands in ceph-ansible master so that we can start testing it in all the Ceph versions.

We were never aware that a merge had to be prioritized with you.

If the CI is taking 4 hours to complete jobs maybe we need to work on thinking a better approach? Until that happens, how can we triage merging our component work? How do we set the priority/severity on PRs?

**Comment by leseb** [comment]
This doesn't have to be a complex mechanism, you can just ping me on IRC about your PR and we can set priorities. Taking ownership of a component does not mean merging without the review of a maintainer.

The CI is a different problem that we are always trying to improve, it's a balance between speed and robustness.

**Comment by alfredodeza** [comment]
We will reach out next time, apologies for the issues this merge caused

**Pull request merged**